### PR TITLE
Make doc2vec imdb ipynb tutorial run in python 2 and 3

### PR DESCRIPTION
**Pull request by robotcator**
fix the compatibility between python2 and python3 for the notebook of doc2vec-IMDB.ipynb  [#1139](https://github.com/RaRe-Technologies/gensim/pull/1139).

### CODE REVIEW

**Comment by tmylk** [comment]
Please merge in develop into your branh to resolve the conflicts `git fetch;git merge develop`

**Comment by robotcator** [comment]
It seems that I select the wrong base branch. I have change to the develop branch and the conflicts were resolved. And the commit( 1aa3f33b7a225bea0b535cb8b8e67f519ddf9772) was my operation mistake.

File: docs/notebooks/doc2vec-IMDB.ipynb [hunk 1]
```diff
@@ -92,8 +116,7 @@
     "        txt_files = glob.glob('/'.join([dirname, fol, '*.txt']))\n",
     "\n",
     "        for txt in txt_files:\n",
-    "            with open(txt, 'r', encoding='utf-8') as t:\n",
-    "                control_chars = [chr(0x85)]\n",
+    "            with codecs.open(txt, 'r', encoding='utf-8') as t:\n",
```
Comment (piskvorky) [inline docs/notebooks/doc2vec-IMDB.ipynb]: Use `smart_open` instead: drop `codecs`, open files in binary mode and convert content to unicode explicitly.

File: docs/notebooks/doc2vec-IMDB.ipynb [hunk 2]
```diff
@@ -104,21 +127,28 @@
     "            temp += \"\\n\"\n",
     "\n",
     "        temp_norm = normalize_text(temp)\n",
-    "        with open('/'.join([dirname, output]), 'w', encoding='utf-8') as n:\n",
+    "        with codecs.open('/'.join([dirname, output]), 'w', encoding='utf-8') as n:\n",
```
Comment (piskvorky) [inline docs/notebooks/doc2vec-IMDB.ipynb]: Not portable -- use `os.path.join`.

File: docs/notebooks/doc2vec-IMDB.ipynb [hunk 3]
```diff
@@ -104,21 +127,28 @@
...
     "            n.write(temp_norm)\n",
     "\n",
     "        alldata += temp_norm\n",
     "\n",
-    "    with open('/'.join([dirname, 'alldata-id.txt']), 'w', encoding='utf-8') as f:\n",
+    "    with codecs.open('/'.join([dirname, 'alldata-id.txt']), 'w', encoding='utf-8') as f:\n",
```
Comment (piskvorky) [inline docs/notebooks/doc2vec-IMDB.ipynb]: Drop `codecs`, use binary mode.

File: docs/notebooks/doc2vec-IMDB.ipynb (hunk 1 above)
Comment (robotcator) [inline docs/notebooks/doc2vec-IMDB.ipynb]: Ok, I will drop the codecs and move to smart_open.

**Pull request merged**